File: android/app/src/main/java/io/perenecabuto/catchcatch/LocationUpdateListener.java
package io.perenecabuto.catchcatch;

import android.location.Location;
import android.location.LocationListener;
import android.os.Bundle;


class LocationUpdateListener implements LocationListener {
    private final Callback callback;

    LocationUpdateListener(Callback callback) {
        this.callback = callback;
    }

    @Override
    public void onLocationChanged(Location location) {
        callback.onUpdate(location);
    }

    @Override
    public void onStatusChanged(String s, int i, Bundle bundle) {

    }

    @Override
    public void onProviderEnabled(String s) {

    }

    @Override
    public void onProviderDisabled(String s) {

    }

    interface Callback {
        void onUpdate(Location l);
    }
}

File: android/app/src/main/java/io/perenecabuto/catchcatch/LocationUpdateListener.kt
package io.perenecabuto.catchcatch

import android.location.Location
import android.location.LocationListener
import android.os.Bundle


internal class LocationUpdateListener(private val callback: Callback) : LocationListener {

    override fun onLocationChanged(location: Location) {
        callback.onUpdate(location)
    }

    override fun onStatusChanged(s: String, i: Int, bundle: Bundle) {
    }

    override fun onProviderEnabled(s: String) {
    }

    override fun onProviderDisabled(s: String) {
    }

    interface Callback {
        fun onUpdate(l: Location)
    }
}
